src/read.js

```diff
@@ -98,11 +102,28 @@ class Entry {
   }
 
   get offset() {
-    return this.dataView.getUint32(42, true);
+    if (this.zip64) {
+      // Read the ZIP64 offset from the extra fields
+      // The extra field 0x0001 is used for ZIP64 information
+      const extraField = this._extraFields[0x0001];
+      if (extraField) {
+        // The offset is the third field in the ZIP64 extra fields
+        return Number(getBigInt64(
+          extraField,
+          (this.compressedSize === MAX_VALUE_32BITS ? 8 : 0) + (this.uncompressedSize === MAX_VALUE_32BITS ? 8 : 0),
+          true));
```

Comment:
Maybe more readable.

```suggestion
        const offset = (this.compressedSize === MAX_VALUE_32BITS ? 8 : 0) + (this.uncompressedSize === MAX_VALUE_32BITS ? 8 : 0);
        return JSBI.toNumber(getBigInt64(extraField, offset, true));
```